deepchecks/vision/checks/performance/confusion_matrix.py

```diff
@@ -72,156 +75,141 @@ def __init__(self,
+
+        confusion_matrix, categories = filter_confusion_matrix(
+            matrix,
+            self.categories_to_display
+        )
+
+        description = [f'Showing {self.categories_to_display} of {dataset.num_classes} classes:']
+        classes_to_display = []
+        classes_map = dict(enumerate(classes))  # class index -> class label
 
-        display_categories = []
-        add_not_found_category = False
-        add_unseen_category = False
         for category in categories:
-            if self.not_found_idx == category:
-                add_not_found_category = True
-            elif self.unseen_class_idx == category:
-                add_unseen_category = True
+            category = classes_map[category]
+            if category == 'no-overlapping':
+                description.append(
+                    '"No overlapping" categories are labels and prediction which did not have a matching '
+                    'label/prediction. For example a predictions that did not have a sufficiently overlapping '
+                    'label bounding box will appear under the "No overlapping label" category'
+                )
+                classes_to_display.append('no-overlapping')
+            elif isinstance(category, int):
+                classes_to_display.append(dataset.label_id_to_name(category))
             else:
-                display_categories.append(self.matrix_id_to_class_name(category, dataset))
-
-        x = display_categories
-        y = x.copy()
-
-        if add_not_found_category:
-            description += ['"No overlapping" categories are labels and prediction which did not have a matching '
-                            'label/prediction. For example a predictions that did not have a sufficiently overlapping '
-                            'label bounding box will appear under the "No overlapping label" category']
-            x += ['No overlapping prediction']
-            y += ['No overlapping label']
-        if add_unseen_category:
-            description += ['Unseen classes are classes that did not appear in the dataset labels, but did were '
-                            'predicted to be present.']
-            x += ['Unseen classes']
-            y += ['Unseen classes']
-
-        description += [f'Showing {self.categories_to_display} of {dataset.num_classes} classes:']
-
-        fig = imshow(display_confusion_matrix,
-                     x=x,
-                     y=y,
-                     text_auto=True)
-
-        fig.update_layout(width=600, height=600)
-        fig.update_xaxes(title='Predicted Value', type='category')
-        fig.update_yaxes(title='True value', type='category')
-
+                raise RuntimeError(
+                    'Internall Error! categories list mmust '
+                    'contain items of type - Union[int, Literal["no-overlapping"]]'
+                )
+
+        # NOTE: 'no-overlapping' is at the end of the list
+        x = classes_to_display[:-1]
```

Comment:
don't you need here some `if 'no-overlapping' in classes_to_display`?

Reply:
fixed